lib/router.js

```diff
@@ -155,33 +201,156 @@ Router.prototype._loadModule = function(modDef, config) {
     }
 };
 
+Router.prototype._loadModules = function(node, restBaseModules, scope) {
+    var self = this;
+    if (Array.isArray(restBaseModules)) {
+        throw new Error('Old style module config detected! '
+                + 'New format expects an object, not an array. '
+                + 'Please update your config.');
+    } else if (typeof restBaseModules === 'object') {
+        return P.each(Object.keys(restBaseModules), function(modulePath) {
+            var pathMods = restBaseModules[modulePath];
+            if (!Array.isArray(pathMods)) {
+                throw new Error('Error in module definition for ' + modulePath + '!\n'
+                        + 'Expected an array, got ' + pathMods);
+            }
+            return P.each(pathMods, function(m) {
+                // Share modules
+                return self._loadModule(m, scope.globals)
```

Comment:
I think now sharing modules has become much more trickier because we'd need to check the options passed to it as well to deduce it's the same mod def, shouldn't we?